### ISSUE TITLE
`lambda expressions are not supported` java `-source` issue

### ISSUE THREAD

**Issue by usernamenumber** [react-native-datetimepicker/datetimepicker#684]
# Versions
- React Native 0.70.4
- React 18.2.0
- rndatepicker 6.5.3

# Question
Apologies in advance if I'm missing something obvious here, I'm still relatively new to android development.

I am getting the following error when I try to build my app:

```
.../node_modules/@react-native-community/datetimepicker/android/src/main/java/com/reactcommunity/rndatetimepicker/Common.java:57: 
  error: lambda expressions are not supported in -source 7                                                                                    
                return presentedDialog -> {                                                           
                                       ^           
  (use -source 8 or higher to enable lambda expressions)                                              
1 error  
```

I think I understand what this means (lambdas aren't supported in java v7, which is being used to build the package), however, my `app/build.gradle` has 

```
    compileOptions {
        sourceCompatibility JavaVersion.VERSION_1_8
        targetCompatibility JavaVersion.VERSION_1_8 }
```

which from what I've read should mean it's using Java v8, so I think there must be some other mechanism by which I can control the `-source` used when compiling datetimepicker.

If someone knows what's going on or what I can do about it, that would be greatly appreciated!

**Comment by vonovak** [comment]
:tada: This issue has been resolved in version 6.5.4 :tada:

If this package helps you, consider [sponsoring us](https://github.com/sponsors/react-native-datetimepicker)! :rocket:

**Comment by jsimonassi** [comment]
I found this same behavior in RNDateTimePickerPackage.java

I opened a PR for this on #753
